src/fides/api/service/connectors/query_configs/query_config.py

```diff
@@ -201,10 +209,44 @@ def update_value_map(  # pylint: disable=R0914
                     )
                 ]
                 for detailed_path in paths_to_mask:
+                    original_value = pydash.objects.get(row, detailed_path)
+
+                    # Skip object fields - we don't want to mask these directly
+                    if isinstance(original_value, dict) and original_value:
+                        logger.debug(f"Skipping object field: {detailed_path}")
+                        continue
+
+                    # Skip arrays of objects, but keep arrays of primitives
+                    if isinstance(original_value, list) and original_value:
+                        if all(isinstance(item, dict) for item in original_value):
+                            logger.debug(
+                                f"Skipping array of objects field: {detailed_path}"
+                            )
+                            continue
```

Comment:
This fixes a bug where complex objects with data categories would be replaced with `null` or string values. That is incorrect, we need to mask individual fields, not the objects themselves.